proto/serverless/instrumentation/tags/v1/tags.proto

```diff
@@ -0,0 +1,290 @@
+syntax = "proto3";
+
+package serverless.instrumentation.tags.v1;
+
+option go_package = ".;protoc";
+
+message Tags {
+  // ============================================ //
+  // Defined TagSets start at field number 100  //
+  // ========================================== //
+
+  // ============================================ //
+  // Top level tags will have their underscores replaced //
+  // with periods when flattened. So for `aws_lambda` it will //
+  // be converted to `aws.lambda`//
```

Comment:
@Mmarzex There's an additional thing we need to acknowledge here.

With the above approach (where in top-level names, we replace `_` with `.`), we lock us from using `_` separator in eventual top-level names to be introduced in the future. While we may never want to have that, if that happens, I'm pretty sure it'll turn quite confusing for developer wanting to define such.

So design-wise I believe it'll be safer and less problematic to not introduce such special handling for top-level tag names, especially that there seems to be not much win with this approach.

Reply:
Maybe for flattening the data, we just have certain root tags allowlisted on the APIs that get flattened. The worst case scenario is that we have `aws_lambda` instead of `aws.lambda` I don't think that is a big deal.